models/message.go

```diff
@@ -108,25 +108,20 @@ func CreateMessage(fields *Message) error {
 // URLs, and then passing the body through Blackfriday for additional parsing
 // and finally Bluemonday for sanitization.
 func ParseMessage(message *Message) string {
-	body := message.Body
+	messageCopy := *message
```

Comment:
@jmingtan, after working through your example and tests, you're right -- these fixes address a bug that I totally missed. I'll work on getting this merged today -- thanks!